[0.7] test_vec2pix and test_vec2ang fail

The Debian package of astropy-healpix got bug report [Debian#1026012](https://bugs.debian.org/1026012), that the build time test now fails. The package was version 0.6, but0.7 shows the same behavior. Specifically, the following two tests fail:
```
_________________________________ test_vec2pix _________________________________

    @given(nside_pow=integers(0, 29), nest=booleans(),
>          x=floats(-1, 1, allow_nan=False, allow_infinity=False).filter(lambda x: abs(x) > 1e-10),
           y=floats(-1, 1, allow_nan=False, allow_infinity=False).filter(lambda y: abs(y) > 1e-10),
           z=floats(-1, 1, allow_nan=False, allow_infinity=False).filter(lambda z: abs(z) > 1e-10))

astropy_healpix/tests/test_healpy.py:116: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

nside_pow = 2, x = 1.0, y = 1.0, z = 0.5, nest = False

    @given(nside_pow=integers(0, 29), nest=booleans(),
           x=floats(-1, 1, allow_nan=False, allow_infinity=False).filter(lambda x: abs(x) > 1e-10),
           y=floats(-1, 1, allow_nan=False, allow_infinity=False).filter(lambda y: abs(y) > 1e-10),
           z=floats(-1, 1, allow_nan=False, allow_infinity=False).filter(lambda z: abs(z) > 1e-10))
    @settings(max_examples=2000, derandomize=True, deadline=None)
    def test_vec2pix(nside_pow, x, y, z, nest):
        nside = 2 ** nside_pow
        ipix1 = hp_compat.vec2pix(nside, x, y, z, nest=nest)
        ipix2 = hp.vec2pix(nside, x, y, z, nest=nest)
>       assert ipix1 == ipix2
E       assert 42 == 58
E       Falsifying example: test_vec2pix(
E           nside_pow=2, nest=False, x=1.0, y=1.0, z=0.5,
E       )

astropy_healpix/tests/test_healpy.py:124: AssertionError
_________________________________ test_vec2ang _________________________________

    @given(vectors=arrays(float, (3,), elements=floats(-1, 1)).filter(not_at_origin),
>          lonlat=booleans(), ndim=integers(0, 4))

astropy_healpix/tests/test_healpy.py:210: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

vectors = array([  1.00000000e+00,  -2.22044605e-16,   1.00000000e+00])
lonlat = False, ndim = 0

    @given(vectors=arrays(float, (3,), elements=floats(-1, 1)).filter(not_at_origin),
           lonlat=booleans(), ndim=integers(0, 4))
    @settings(max_examples=500, derandomize=True, deadline=None)
    def test_vec2ang(vectors, lonlat, ndim):
        vectors = np.broadcast_to(vectors, (2,) * ndim + (3,))
        theta1, phi1 = hp_compat.vec2ang(vectors, lonlat=lonlat)
        theta2, phi2 = hp.vec2ang(vectors, lonlat=lonlat)
        # Healpy sometimes returns NaNs for phi (somewhat incorrectly)
        phi2 = np.nan_to_num(phi2)
        assert_allclose(theta1, theta1, atol=1e-10)
>       assert_allclose(phi1, phi2, atol=1e-10)
E       AssertionError: 
E       Not equal to tolerance rtol=1e-07, atol=1e-10
E       
E       Mismatched elements: 1 / 1 (100%)
E       Max absolute difference: 6.283185307179586
E       Max relative difference: 1.0
E        x: array([ 0.])
E        y: array([ 6.283185])
E       Falsifying example: test_vec2ang(
E           vectors=array([  1.00000000e+00,  -2.22044605e-16,   1.00000000e+00]),
E           lonlat=False,
E           ndim=0,
E       )

astropy_healpix/tests/test_healpy.py:219: AssertionError
```
This is with 

* Python 3.10.9, 3.11.1
* Astropy 5.1.1 (bug report), 5.2 (updated test)
* Numpy: 1.23.5
* Scipy: 1.8.1
* Matplotlib: 3.5.2

The package was [reported to pass the integration testing for Astropy 5.2](https://github.com/astropy/astropy/wiki/v5.2-RC-Testing).

--- a/.github/workflows/ci_tests.yml
+++ b/.github/workflows/ci_tests.yml
@@ -42,10 +42,10 @@ jobs:
             python: '3.11'
             toxenv: py311-test-alldeps
 
-          - name: Windows - Python 3.11 with all optional dependencies
+          - name: Windows - Python 3.11 with minimal dependencies
             os: windows-latest
             python: '3.11'
-            toxenv: py311-test-alldeps
+            toxenv: py311-test
 
           - name: Python 3.12 with all optional dependencies and pre-releases
             os: ubuntu-latest

--- a/astropy_healpix/tests/test_healpy.py
+++ b/astropy_healpix/tests/test_healpy.py
@@ -1,12 +1,15 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 from itertools import product
 
+from astropy.coordinates import angular_separation
+from astropy import units as u
 import pytest
 import numpy as np
 
 from numpy.testing import assert_equal, assert_allclose
 
 from .. import healpy as hp_compat
+from ..high_level import xyz_to_healpix
 
 from hypothesis import given, settings, example
 from hypothesis.strategies import integers, floats, booleans
@@ -119,9 +122,14 @@ def test_pix2ang(nside_pow, frac, nest, lonlat):
 @settings(max_examples=2000, derandomize=True, deadline=None)
 def test_vec2pix(nside_pow, x, y, z, nest):
     nside = 2 ** nside_pow
-    ipix1 = hp_compat.vec2pix(nside, x, y, z, nest=nest)
-    ipix2 = hp.vec2pix(nside, x, y, z, nest=nest)
-    assert ipix1 == ipix2
+    _, dx, dy = xyz_to_healpix(
+        x, y, z, nside, return_offsets=True, order="nested" if nest else "ring"
+    )
+    # Skip vectors that are on the boundary of a pixel where ipix is ambiguous
+    if 0 < dx < 1 and 0 < dy < 1:
+        ipix1 = hp_compat.vec2pix(nside, x, y, z, nest=nest)
+        ipix2 = hp.vec2pix(nside, x, y, z, nest=nest)
+        assert ipix1 == ipix2
 
 
 @given(nside_pow=integers(0, 29), nest=booleans(),
@@ -211,12 +219,15 @@ def not_at_origin(vec):
 @settings(max_examples=500, derandomize=True, deadline=None)
 def test_vec2ang(vectors, lonlat, ndim):
     vectors = np.broadcast_to(vectors, (2,) * ndim + (3,))
-    theta1, phi1 = hp_compat.vec2ang(vectors, lonlat=lonlat)
-    theta2, phi2 = hp.vec2ang(vectors, lonlat=lonlat)
-    # Healpy sometimes returns NaNs for phi (somewhat incorrectly)
-    phi2 = np.nan_to_num(phi2)
-    assert_allclose(theta1, theta1, atol=1e-10)
-    assert_allclose(phi1, phi2, atol=1e-10)
+    lon1, lat1 = hp_compat.vec2ang(vectors, lonlat=lonlat)
+    lon2, lat2 = hp.vec2ang(vectors, lonlat=lonlat)
+    if not lonlat:
+        lon1, lat1 = np.rad2deg(lat1), 90 - np.rad2deg(lon1)
+        lon2, lat2 = np.rad2deg(lat2), 90 - np.rad2deg(lon2)
+    sep = angular_separation(
+        lon1 * u.deg, lat1 * u.deg, lon2 * u.deg, lat2 * u.deg
+    ).to_value(u.rad)
+    assert_allclose(sep, 0, atol=1e-8)
 
 
 @given(lonlat=booleans(),

--- a/astropy_healpix/tests/test_healpy.py
+++ b/astropy_healpix/tests/test_healpy.py
@@ -12,7 +12,7 @@ from .. import healpy as hp_compat
 from ..high_level import xyz_to_healpix
 
 from hypothesis import given, settings, example
-from hypothesis.strategies import integers, floats, booleans
+from hypothesis.strategies import integers, floats, booleans, tuples
 from hypothesis.extra.numpy import arrays
 
 # NOTE: If healpy is installed, we use it in these tests, but healpy is not a
@@ -115,21 +115,31 @@ def test_pix2ang(nside_pow, frac, nest, lonlat):
             assert_allclose(phi1, phi2, atol=1e-10)
 
 
-@given(nside_pow=integers(0, 29), nest=booleans(),
-       x=floats(-1, 1, allow_nan=False, allow_infinity=False).filter(lambda x: abs(x) > 1e-10),
-       y=floats(-1, 1, allow_nan=False, allow_infinity=False).filter(lambda y: abs(y) > 1e-10),
-       z=floats(-1, 1, allow_nan=False, allow_infinity=False).filter(lambda z: abs(z) > 1e-10))
-@settings(max_examples=2000, derandomize=True, deadline=None)
-def test_vec2pix(nside_pow, x, y, z, nest):
+def not_on_boundaries(args):
+    """Skip vectors that are on the boundary of a pixel where ipix is ambiguous."""
+    nside_pow, nest, x, y, z = args
     nside = 2 ** nside_pow
     _, dx, dy = xyz_to_healpix(
         x, y, z, nside, return_offsets=True, order="nested" if nest else "ring"
     )
-    # Skip vectors that are on the boundary of a pixel where ipix is ambiguous
-    if 0 < dx < 1 and 0 < dy < 1:
-        ipix1 = hp_compat.vec2pix(nside, x, y, z, nest=nest)
-        ipix2 = hp.vec2pix(nside, x, y, z, nest=nest)
-        assert ipix1 == ipix2
+    return 0 < dx < 1 and 0 < dy < 1
+
+
+@given(
+    args=tuples(
+        integers(0, 29), booleans(),
+        floats(-1, 1, allow_nan=False, allow_infinity=False).filter(lambda x: abs(x) > 1e-10),
+        floats(-1, 1, allow_nan=False, allow_infinity=False).filter(lambda y: abs(y) > 1e-10),
+        floats(-1, 1, allow_nan=False, allow_infinity=False).filter(lambda z: abs(z) > 1e-10)
+    ).filter(not_on_boundaries)
+)
+@settings(max_examples=2000, derandomize=True, deadline=None)
+def test_vec2pix(args):
+    nside_pow, nest, x, y, z = args
+    nside = 2 ** nside_pow
+    ipix1 = hp_compat.vec2pix(nside, x, y, z, nest=nest)
+    ipix2 = hp.vec2pix(nside, x, y, z, nest=nest)
+    assert ipix1 == ipix2
 
 
 @given(nside_pow=integers(0, 29), nest=booleans(),

--- a/tox.ini
+++ b/tox.ini
@@ -50,6 +50,9 @@ deps =
     devdeps: numpy>=0.0.dev0
     devdeps: pyerfa>=0.0.dev0
     devdeps: astropy>=0.0.dev0
+    devdeps: healpy
+
+    alldeps: healpy
 
 # The following indicates which extras_require from setup.cfg will be installed
 extras =

